components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java

```diff
@@ -19,10 +19,548 @@
+
+    @Override
+    public SiddhiTopology createTopology(String userDefinedSiddhiApp) {
+        this.siddhiApp = SiddhiCompiler.parse(userDefinedSiddhiApp);
+        this.siddhiAppRuntime = (new SiddhiManager()).createSiddhiAppRuntime(userDefinedSiddhiApp);
+        SiddhiQueryGroup siddhiQueryGroup;
+        int[] queryContextEndIndex;
+        int[] queryContextStartIndex;
+        String execGroupName;
+        int parallel;
+
+
+
+        this.siddhiTopologyDataHolder = new SiddhiTopologyDataHolder(getAppName(), userDefinedSiddhiApp);
+
+        for (ExecutionElement executionElement : siddhiApp.getExecutionElementList()) {
+            parallel = 1;
+            execGroupName = null;
+
+            for (int i = 0; i < executionElement.getAnnotations().size(); i++) {
+                if (executionElement.getAnnotations().get(i).getElement("execGroup") != null) {
+                    execGroupName = siddhiTopologyDataHolder.getSiddhiAppName() + "-" + executionElement.getAnnotations().get(i).getElement("execGroup");
+
+                }
+
+                if (executionElement.getAnnotations().get(i).getElement("parallel") != null) {
+                    parallel = Integer.parseInt(executionElement.getAnnotations().get(i).getElement("parallel"));
+
+                }
+            }
+            if (execGroupName != null && !siddhiTopologyDataHolder.getSiddhiQueryGroupMap().containsKey(execGroupName)) {
+
+                siddhiQueryGroup = new SiddhiQueryGroup();
+                siddhiQueryGroup.setName(execGroupName);
+                siddhiQueryGroup.setParallelism(parallel);
+
+            } else if (siddhiTopologyDataHolder.getSiddhiQueryGroupMap().containsKey(execGroupName)) {
+                siddhiQueryGroup = siddhiTopologyDataHolder.getSiddhiQueryGroupMap().get(execGroupName);
+
+                //Same execution group given  with different parallel numbers
+                if (siddhiQueryGroup.getParallelism() != parallel) {
+                    throw new SiddhiAppValidationException("execGroup =" + "\'" + execGroupName + "\' not assigned unique @dist(parallel)");
+                }
+            } else {
+                //will work if execGroup is not mentioned-those will go to a single app
+                siddhiQueryGroup = new SiddhiQueryGroup();
+                siddhiQueryGroup.setName(siddhiTopologyDataHolder.getSiddhiAppName() + UUID.randomUUID());
+                siddhiQueryGroup.setParallelism(parallel);
+            }
+            //if execution element is a query
+            if (executionElement instanceof Query) {
+
+                //set query
+                queryContextStartIndex = ((Query) executionElement).getQueryContextStartIndex();
+                queryContextEndIndex = ((Query) executionElement).getQueryContextEndIndex();
+                siddhiQueryGroup.addQuery(removeMetainfoQuery(executionElement, ExceptionUtil.getContext(queryContextStartIndex, queryContextEndIndex, userDefinedSiddhiApp)));
+
```

Comment:
I think these new lines(2) are unnecessary. Can you please remove them? Please check other classes also